Tests/Common/Data/Auxiliary/AuxiliaryDataSerializationTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Tests.Common.Data.Auxiliary
+{
+    [TestFixture]
+    public class AuxiliaryDataSerializationTests
+    {
+        private readonly JsonSerializerSettings _settings = new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All };
+
+        [Test]
+        public void DeserializesSplitWarning()
+        {
+            var splitWarning = new Split(Symbols.AAPL, new DateTime(2014, 6, 9), 645.57m, 0.142857m, SplitType.Warning);
+
+            var json = JsonConvert.SerializeObject(splitWarning, _settings);
+            var deserialized = (Split)JsonConvert.DeserializeObject(json, _settings);
```

Comment:
no need to change, can also use syntax like `JsonConvert.DeserializeObject<Split>` instead of using the direct cast.